src/Accordion.js

```diff
@@ -0,0 +1,99 @@
+import React, { cloneElement, useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const Accordion = ({
+    children,
+    className,
+    closedClassName,
+    defaultActiveItem,
+    icon,
+    iconClassName,
+    onClosing,
+    onOpening,
+    onToggle,
+    multiple,
+    scrollIntoView,
+}) => {
+    const [activeIndex, setActiveIndex] = useState(() => {
+        switch (true) {
+        case defaultActiveItem != null:
+            if (multiple) {
+                return Array.isArray(defaultActiveItem) ? defaultActiveItem : [defaultActiveItem];
+            }
+
+            return Array.isArray(defaultActiveItem) ? defaultActiveItem[0] ?? null : defaultActiveItem;
+
+        default:
+            return multiple ? [] : null;
+        }
+    });
+
+    const setOpenItem = useCallback((index) => {
+        switch (true) {
+        case multiple:
+            if (activeIndex.includes(index)) {
+                onClosing && onClosing(index);
+                setActiveIndex(activeIndex.filter((i) => i !== index));
+            } else {
+                onOpening && onOpening(index);
+                setActiveIndex([...activeIndex, index].sort());
+            }
+            break;
+
+        default:
+            if (activeIndex === index) {
+                onClosing && onClosing();
+                setActiveIndex(null);
+            } else {
+                onOpening && onOpening(index);
+                setActiveIndex(index);
+            }
+            break;
+        }
+    }, [activeIndex, multiple, onClosing, onOpening]);
+
+    const renderItem = (child, index) => cloneElement(child, {
```

Comment:
This function was a good use in a class component, since it was only defined once. In a functional we are creating this in every render.

If we stick with function, can we move it to the  `childrenArray.map` directly?